models/bookmark.go

```diff
@@ -1,8 +1,8 @@
 package models
 
 type Bookmark struct {
-	UserRefer  uint     `gorm: not null; default:null`
-	UserId     Users    `gorm:"foreignKey:UserRefer"`
+	UserRefer  Users     `gorm:"foreignKey:UserRefer"`

```

Comment:
It's self referencing.

In previous tables UserRefer is the uint (key) and the Users Users `gorm:"foreignKey:UserRefer"` as the reference

Look at the Location model.